.github/workflows/build_and_run_tests.yml

```diff
@@ -59,7 +59,23 @@ jobs:
         ./meson.py --buildtype=release build.release
         ./meson.py --buildtype=debug build.asan -Db_sanitize=address
     - name: Build and run release tests
+      id: buildnormal
       run: ./meson.py test -C build.release
 
+    - name: Save error logs
+      uses: actions/upload-artifact@v3
+      if: steps.buildnormal.outcome == 'failure'
```

Comment:
Hmm, just reading the docs of the expression [failure()](https://docs.github.com/en/actions/learn-github-actions/expressions#failure) that implies to have an equivalent construct: `if: ${{ failure() }}`. Isn't that more suitable?

Reply:
That is triggered on ANY failure, for example when fetching Meson failed :grimacing: 
Also the differentiation between the two builds is then lost.